src/main/scala/cromwell/engine/io/gcs/GcsFileSystem.scala

```diff
@@ -0,0 +1,61 @@
+package cromwell.engine.io.gcs
+
+import java.lang.Iterable
+import java.nio.file._
+import java.nio.file.attribute.UserPrincipalLookupService
+import java.nio.file.spi.FileSystemProvider
+import java.util
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable.HashSet
+import scala.concurrent.ExecutionContext
+import scala.util.Try
+
+object GcsFileSystem {
+  def getInstance(provider: GcsFileSystemProvider) = new GcsFileSystem(provider)
+  def getInstance(interface: GoogleCloudStorage)(implicit executionContext: ExecutionContext) = new GcsFileSystem(GcsFileSystemProvider.getInstance(interface))
+}
+
+class GcsFileSystem(gcsFileSystemProvider: GcsFileSystemProvider) extends FileSystem {
+
+  private val SEPARATOR = "/"
+  private [io] val PROTOCOL = "gs://"
+  private val gsUriRegex = s"""$PROTOCOL(.*)""".r
+
+  override def supportedFileAttributeViews(): util.Set[String] = HashSet("basic").asJava
```

Comment:
Maybe just make a Java Set directly?  Google Collections has some sugar for initialization.